### PR TITLE
fix: adding support for other data types in orjson_dumps

### PR DESCRIPTION
**Pull request by AndreaFrancis**
Part of https://github.com/huggingface/datasets-server/issues/1443
Currently, we have 105 cache records with error types like:

```
  'Type is not JSON serializable: Timedelta',
  'Type is not JSON serializable: Timestamp',
  'Type is not JSON serializable: datetime.timedelta',
  'Type is not JSON serializable: numpy.int64',
  'Type is not JSON serializable: numpy.ndarray'
```

The problem is at https://github.com/huggingface/datasets-server/blob/main/libs/libcommon/src/libcommon/utils.py#L120 trying to serialize the values.

After this PR, I will force refresh the affected splits (most of them are for `split-first-rows-from-streaming`)

Also there is an error for "split-descriptive-statistics":

`"Dict key must be str"
`
This was due to non-string keys in ClassLabels like ints. After this PR, I will force refresh the affected splits (49 records)

### CODE REVIEW

File: libs/libcommon/src/libcommon/utils.py [hunk 1]
```diff
@@ -113,7 +114,13 @@ def orjson_default(obj: Any) -> Any:
         # the bytes are encoded with base64, and then decoded as utf-8
         # (ascii only, by the way) to get a string
         return base64.b64encode(obj).decode("utf-8")
-    raise TypeError
+    if isinstance(obj, np.integer):
+        return int(obj)
+    if isinstance(obj, np.floating):
+        return float(obj)
+    if isinstance(obj, np.ndarray):
+        return obj.tolist()
```
Comment (albertvillanova) [inline libs/libcommon/src/libcommon/utils.py:122]: To support the serialization of `numpy` types, why don't you use the `orjson.OPT_SERIALIZE_NUMPY` option in `orjson.dumps`?

**Review by albertvillanova — APPROVED** [review]
Thanks. Just a question below.

File: libs/libcommon/src/libcommon/utils.py (hunk 1 above)
Comment (AndreaFrancis) [reply libs/libcommon/src/libcommon/utils.py:122]: Thanks for the suggestion! I will send a new commit with this option.

**Review by lhoestq — APPROVED** [review]
Nice !

The datetime formatting is like `b'"2022-01-15T00:00:00"'` and numpy arrays is like `b'[0,1,2,3,4]'` which is perfect

**Comment by lhoestq** [comment]
Cc @severo in case it requires changes in the front-end (might work out of the box no?)

**Comment by lhoestq** [comment]
This should fix https://github.com/huggingface/datasets-server/issues/86 btw :)

**Comment by severo** [comment]
By default, it will return a string. It should work

**Review by albertvillanova — COMMENTED** [review]
Just another comment: the datetime and the Timestamp are not currently aligned
- datetime: 
  ```
  b'"2017-01-01T12:10:20.345000"'
  ```
- Timestamp: 
  ```
  b'"2017-01-01 12:10:20.345000"'
  ```

If the alignment is important, this could be achieved with:
```python
def orjson_default(obj):
    ...
    if isinstance(obj, pd.Timestamp):
        return obj.to_pydatetime()
```

**Comment by AndreaFrancis** [comment]
> Just another comment: the datetime and the Timestamp are not currently aligned

Thanks @albertvillanova! I added your suggestion.

**Comment by AndreaFrancis** [comment]
Fixed:
```
Atlas atlas-x5jgb3-shard-0 [primary] datasets_server_cache> db.cachedResponsesBlue.countDocuments({"details.copied_from_artifact":{$exists:false}, error_code:"UnexpectedError", "content.error":/Dict key must be str/i})
0
Atlas atlas-x5jgb3-shard-0 [primary] datasets_server_cache> db.cachedResponsesBlue.countDocuments({"details.copied_from_artifact":{$exists:false}, error_code:"UnexpectedError", "content.error":/Type is not JSON serializable/i})
0

```

**Comment by severo** [comment]
you rock!

**Pull request merged**